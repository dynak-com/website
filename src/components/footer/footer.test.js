import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Footer from './footer.component';
import useComponent from './hook';

describe('Footer component', () => {
    const component = shallow(<Footer />);
    const wrapper = mount(<Footer />);

    it('should render 2 spans', () => {
        const footer = component.find('footer');
        expect(footer.find('span')).toHaveLength(2);
    });

    it('displays RODO text', () => {
        const rodoSign = wrapper.find('.rodo');
        expect(rodoSign.text()).toEqual('RODO');
    });

    it('renders correctly', () => {
        const tree = renderer.create(<Footer />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

function HookWrapper(props) {
    const hook = props.hook ? props.hook() : undefined;
    return <HookWrapper hook={hook} />;
}

describe('Footer hook', () => {
    const component = shallow(<Footer />);
    it('renders correctly', () => {
        expect(component.exists()).toBeTruthy();
    });

    it('should set init value', () => {
        const wrapper = shallow(<HookWrapper hook={() => useComponent()} />);

        const hook = wrapper.props().hook;
        const currentYear = new Date().getFullYear();

        const { year } = hook;
        expect(year).toEqual(currentYear);

        const property = component.find('span').at(1).prop('onClick');
        expect(typeof property === 'function' ? true : false).toBeTruthy();
    });
});
