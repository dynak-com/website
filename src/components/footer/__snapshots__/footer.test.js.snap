// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders correctly 1`] = `
<footer
  className="rodoContainer my-2 text-light text-center mockyApp"
  data-jest="mockyApp"
>
  <span
    className="rodoComp"
  >
    Â©dynak.com 
    2020
  </span>
  <span
    className="rodo custom-styled-span ml-2"
    onClick={[Function]}
  >
    RODO
  </span>
</footer>
`;
